utils/preferences/src/preferences.rs

```diff
@@ -0,0 +1,30 @@
+use icu_locid::{
+    subtags::{Language, Region, Script},
+    Locale,
+};
+
+pub trait Preferences {
```

Comment:
Nit: In this new model, I think the value of a standalone `Preferences` trait has gone down significantly. I think we should get rid of it and just `impl From<X> for DateTimeFormatPreferences` for all `X` we care about.

Reply:
I think the `Preferences` here will eventually fold into `DataLocale`. But I removed it for now.